### PR TITLE
Add PersonDirectory sample code

### PR DESCRIPTION
**Pull request by zhizhoualan**

### CODE REVIEW

**Comment by der3318** [comment]
Wondering if it will be better to name the sample as PersonDirectoryOperations or PersonDirectoryOps to be clearer?

File: face/PersonDirectory/python/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+## Contents
+| Notebook | Description | Enrollment | Identification |  
+|----------|-------------|------------|----------------|
+| [Enroll persons with single face](enroll_single_face_per_person.ipynb) | The application enrolls one face for each person from a directory of images, with each image containing a different person. | Single face per person | Against a dnynamic person group/a specific list of persons |
+| [Enroll persons with multiple faces](enroll_multiple_faces_per_person.ipynb) | The application enrolls multiple faces for each person from a directory where each subfolder represents a different person. | Mutiple faces per person | Against a dnynamic person group/a specific list of persons |
+
+## Shared Functions
+
+For convenience, commonly used functions across these notebooks are consolidated in [shared_functions.ipynb](shared_functions.py). Import these functions in any notebook as needed.
```
Comment (der3318) [inline face/PersonDirectory/python/README.md:17]:
```suggestion
For convenience, commonly used functions across these notebooks are consolidated in [shared_functions.py](shared_functions.py). Import these functions in any notebook as needed.
```

File: face/PersonDirectory/python/shared_functions.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
+import requests, time
+from PIL import Image
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.vision.face import FaceClient
+from azure.ai.vision.face.models import FaceDetectionModel, FaceRecognitionModel
+
+def detect_faces(subscription_key, endpoint, image_path, injection_header=None):
```
Comment (der3318) [inline face/PersonDirectory/python/shared_functions.py:7]: None of the callers specify injection_header. Is this a miss?

File: face/PersonDirectory/python/enroll_multiple_faces_per_person.ipynb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    "print(f\"Person IDs: {persond_ids}\")\n",
+    "# Identify faces in the test image using the person IDs\n",
+    "identify_faces(FACE_KEY, FACE_ENDPOINT, test_image_path, person_ids=persond_ids, injection_header=\"sample=enroll_multiple_faces_per_person\")\n",
+    "\n",
+    "# Create a dynamic person group based on the person IDs\n",
+    "if create_dynamic_person_group(FACE_KEY, FACE_ENDPOINT, dynamic_person_group_id, persond_ids, \"sample=enroll_multiple_faces_per_person\"):\n",
+    "    print(f\"Created dynamic person group with ID: {dynamic_person_group_id}\")\n",
+    "\n",
+    "    # Identify faces in the test image using the dynamic person group\"\"\"\n",
+    "    identify_faces(FACE_KEY, FACE_ENDPOINT, test_image_path, dynamic_person_group_id=dynamic_person_group_id, injection_header=\"sample=enroll_multiple_faces_per_person\")"
```
Comment (der3318) [inline face/PersonDirectory/python/enroll_multiple_faces_per_person.ipynb:66]: Do we want to clean up the DPG and the person entries at the end of the samples?

File: face/PersonDirectory/python/shared_functions.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    elif dynamic_person_group_id:
+        body['dynamicPersonGroupId'] = dynamic_person_group_id
+    else:
+        print("Person IDs or Dynamic Person Group ID must be provided.")
+        return
+
+    response = requests.post(identify_url, headers=headers, json=body)
+    response.raise_for_status()
+    results = response.json()
+    for i, result in enumerate(results):
```
Comment (der3318) [inline face/PersonDirectory/python/shared_functions.py:153]: Your identification call takes a single face ID:

```python
    body = {
>>      'faceIds': [face_id],
        'maxNumOfCandidatesReturned': 1,
        'confidenceThreshold': 0.5
    }
```

What's the motivation to enumerate the results here?

Comment (zhizhoualan) [reply face/PersonDirectory/python/shared_functions.py]: Good catch. All detected faces are used in identification.

File: face/PersonDirectory/python/enroll_multiple_faces_per_person.ipynb (hunk 3 above)
Comment (zhizhoualan) [reply face/PersonDirectory/python/enroll_multiple_faces_per_person.ipynb]: Good suggestion. Added as the option.

File: face/PersonDirectory/python/shared_functions.py (hunk 2 above)
Comment (zhizhoualan) [reply face/PersonDirectory/python/shared_functions.py]: You are right. Injection is added.

**Review by der3318 — APPROVED** [review]

**Pull request merged**